### PR TITLE
PLATUI-1368 Routes fix

### PR DESCRIPTION
**Pull request by wolfendale**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/nunjucks/NunjucksHelperModule.scala [hunk 1]
```diff
@@ -51,6 +53,35 @@ object NunjucksHelper {
 
   val className = "_nunjucksHelper"
 
+  private val routesInitScript =
```
Comment (matthewmascord) [inline src/main/scala/uk/gov/hmrc/nunjucks/NunjucksHelperModule.scala:56]: Would it be worth looking at potential performance implications of this new mechanism? Does this get evaluated on every page render?

Comment (wolfendale) [reply src/main/scala/uk/gov/hmrc/nunjucks/NunjucksHelperModule.scala:56]: That's already the case. The performance will definitely be slightly worse, but it'll basically be no different for services with fewer than 100 routes anyway.

File: src/main/scala/uk/gov/hmrc/nunjucks/NunjucksHelperModule.scala [hunk 2]
```diff
@@ -107,13 +136,26 @@ object NunjucksHelper {
...
+      Context.exit()
+      scope
+    }
+
+    cx.evaluateString(scope, routesInitScript, "init_routes", 0, null)
+
+    // we need to batch routes as trireme fails to run the script if it's too large
+    routesHelper.routes.sliding(100).foreach { batch =>
+      val script = JavaScriptReverseRouter("batch")(batch: _*)(request).toString
+      cx.evaluateString(scope, s"(function () { $script; mergeDeep(routes, batch); })();", "batch", 0, null);
```
Comment (matthewmascord) [inline src/main/scala/uk/gov/hmrc/nunjucks/NunjucksHelperModule.scala:155]: Is there anything we can do around assuring the output of the successive mergeDeep calls is the same as the previous implementation?

**Comment by oscarduignan** [comment]
@wolfendale and @matthewmascord is it worth having a note in the README describing the issue and say if people update and run into any problems if they're not impacted by the original issue this fixes they can downgrade the version safely?

**Pull request merged**